x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/slm/SnapshotLifecycleServiceTests.java

```diff
@@ -206,13 +207,20 @@ public void testPolicyCRUD() throws Exception {
             sls.clusterChanged(event);
             assertThat(sls.getScheduler().scheduledJobIds(), equalTo(Collections.singleton("foo-2")));
 
+            CopyOnWriteArrayList<String> triggeredJobs = new CopyOnWriteArrayList<>();
             trigger.set(e -> {
-                // Make sure the job got updated
-                assertThat(e.getJobName(), equalTo("foo-2"));
+                triggeredJobs.add(e.getJobName());
                 triggerCount.incrementAndGet();
             });
             clock.fastForwardSeconds(1);
 
+            // Let's make sure the job got updated
+            // We don't simply assert that triggeredJobs has one element with value "foo-2" because of a race condition that can see the
+            // list containing <foo-2, foo-1>. This can happen because when we update the policy to version 2 (ie. to foo-2) we will
+            // cancel the existing policy (foo-1) without waiting for the thread executing foo-1 to actually interrupt
+            // (see org.elasticsearch.common.util.concurrent.FutureUtils#cancel) which means that foo-1 could actually get to be
+            // rescheduled and re-run before it is indeed cancelled.
+            assertBusy(() -> assertThat(triggeredJobs, contains("foo-2")));
```

Comment:
This will eventually fail, the `contains` matcher requires exact matching; so it'll fail and complain with:

```
java.lang.AssertionError: 
Expected: iterable containing ["foo-2"]
     but: item 0: was "foo-1"
```

It should be changed to check that it has `"foo-2"`, but can have anything else also.

Reply:
Ah, that's correct (and why it hadn't fail locally - `contains` is very much overloaded) 

Thanks @dakrone , I pushed a fix to use `hasItem`